### PR TITLE
Update sample to use latest version of Event Hubs package

### PR DESCRIPTION
**Pull request by ramya-rao-a**
## Purpose
The current sample uses an older version of the `@azure/event-hubs` package. This PR updates the sample to use the latest version of the same package which has newer APIs

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x] Other.: Uses latest version of the `@azure/event-hubs` package that has newer APIs
```

## How to Test
*  Get the code

```
git clone https://github.com/Azure-Samples/azure-iot-samples-node
cd azure-iot-samples-node
git checkout [branch-name]
cd iot-hub/QuickStarts/read-d2c-messages
npm install
```

* Test the code
```
Add your event hub compatible endpoint information by running Azure CLI, and run the code
Send data to your IotHub via any means you are familiar with in another program
```

## What to Check
Verify that the sample prints out messages that you have sent
* ...

### CODE REVIEW

**Comment by ramya-rao-a** [comment]
cc @elhorton, @jebrando, @JimacoMS3, @robinsh, @philmea

**Comment by ramya-rao-a** [comment]
The corresponding Quick Start doc update can be found in https://github.com/MicrosoftDocs/azure-docs-pr/pull/112559

**Comment by ramya-rao-a** [comment]
cc @chradek

File: iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js [hunk 1]
```diff
@@ -1,55 +1,87 @@
...
+
+  For an example that uses checkpointing, follow up this sample with the sample in the 
+  eventhubs-checkpointstore-blob package on GitHub at the following link:
+
+  https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/eventhubs-checkpointstore-blob/samples/javascript/receiveEventsUsingCheckpointStore.js
+*/
+
+const { EventHubConsumerClient } = require("@azure/event-hubs");
+
+// If using websockets, uncomment the below require statement
```
Comment (JimacoMS3) [inline iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js:17]: "the below require statement" -> "the require statement below"

File: iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js [hunk 2]
```diff
@@ -1,55 +1,87 @@
...
-}).then(function (ids) {
-  console.log("The partition ids are: ", ids);
-  return ids.map(function (id) {
-    return ehClient.receive(id, printMessage, printError, { eventPosition: EventPosition.fromEnqueuedTime(Date.now()) });
+async function main() {
+  console.log("IoT Hub Quickstarts - Read device to cloud messages.");
+
+  // Event Hub-compatible endpoint
+  // az iot hub show --query properties.eventHubEndpoints.events.endpoint --name {your IoT Hub name}
+  const s_eventHubsCompatibleEndpoint = "{your Event Hubs compatible endpoint}";
```
Comment (JimacoMS3) [inline iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js:49]: Nit 1: Is there anyway to move these const statements to the top of the file? If that's bad form never mind or if you'd rather keep them here that's fine too. (They won't be hard to find.)

Nit 2: are the constant names compatible with Node naming conventions? (I don't know, I'm just asking.) I can always change the doc page if they aren't and you rename them.

PS: both of these are NITS. Totally up to you wrt what is best.

**Comment by JimacoMS3** [comment]
@ramya-rao-a Left a couple of comments.. The one about constant names and location is totally up to you. They are just nits.  I really like that you included code and instructions for Web sockets! I sign-off.

File: iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js (hunk 2 above)
Comment (ramya-rao-a) [reply iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js:49]: Yes, they can be moved up, not a problem.
No, the names dont follow the Node naming conventions :( We like camelCasing
I followed this as you asked to match your quick start update.

What does `s_` even mean??

Comment (JimacoMS3) [reply iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js:49]: Cool. If you would move them up, that would be great. I have no idea what "s_" means, we can probably drop that :)  Feel free to change the constants to whatever casing and names make sense and I'll change the docs to match.

File: iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js (hunk 1 above)
Comment (ramya-rao-a) [reply iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js]: addressed in https://github.com/Azure-Samples/azure-iot-samples-node/pull/39/commits/058d62c74ce7afa6965a8dc09eced55af9e4a340

File: iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js (hunk 2 above)
Comment (ramya-rao-a) [reply iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js]: addressed in https://github.com/Azure-Samples/azure-iot-samples-node/pull/39/commits/058d62c74ce7afa6965a8dc09eced55af9e4a340

**Comment by ramya-rao-a** [comment]
Sounds good @JimacoMS3 

Any idea who to ping to get the PR approved and merged?

**Comment by JimacoMS3** [comment]
@robinsh, @jebrando , @elhorton -- I assume one of you has powers. Can you please merge this PR and the linked PR? Updates to the docs are already in-flight!

**Pull request merged**